src/main/java/com/amplitude/Amplitude.java

```diff
@@ -136,15 +138,27 @@ public synchronized void flushEvents() {
       List<Event> eventsInTransit = new ArrayList<>(eventsToSend);
       eventsToSend.clear();
       CompletableFuture.supplyAsync(
-          () -> {
-            Response response = httpCall.syncHttpCallWithEventsBuffer(eventsInTransit);
-            Status status = response.status;
-
-            if (Retry.shouldRetryForStatus(status)) {
-              Retry.sendEventsWithRetry(eventsInTransit, response, httpCall);
-            }
-            return null;
-          });
+              () -> {
+                Response response = null;
+                try {
+                  response = httpCall.syncHttpCallWithEventsBuffer(eventsInTransit);
+                } catch (AmplitudeInvalidAPIKeyException e) {
+                  throw new CompletionException(e);
+                }
+                return response;
+              })
+          .thenAcceptAsync(
+              response -> {
+                Status status = response.status;
+                if (Retry.shouldRetryForStatus(status)) {
+                  Retry.sendEventsWithRetry(eventsInTransit, response, httpCall);
+                }
+              })
+          .exceptionally(
+              exception -> {
+                logger.error("Invalid API Key", exception.getMessage());
+                return null;
```

Comment:
Would we consider throw the exception here? I guess for now user wouldn't get the exception?